### PR TITLE
[Configuration] Adding mnc2bids converter configuration settings

### LINKED ISSUE #7477: Add a new config section for the BIDS constant variables of the mnc2bids converter script
This was discussed at the imaging meeting of June 11th, 2021.

In the [LORIS-MRI PR #636](https://github.com/aces/Loris-MRI/pull/636), there are some constant values present at the top of the script: https://github.com/aces/Loris-MRI/pull/636/files#r649561354.

Need to create Config settings for these constants so they can be filled on the frontend by the user.

List of new config settings needed for the mnc2bids converter script (they should all go under a new section called `MINC to BIDS converter` to avoid confusions with the rest of the imaging pipeline settings):
- `bids_dataset_authors`: should be an array in the config with one line per author
- `bids_acknowledgments_text`: should be a text
- `bids_readme_text`: should be a text
- `bids_validator_options_to_ignore`: should be an array

### PR DESCRIPTION
**Pull request by AlexandraLivadas**
Includes an SQL patch and changes to `SQL/0000-00-03-ConfigTables.sql` to add configuration settings to the Configuration module for the MINC to BIDS converter script
**To be discussed:** Descriptions and labels for each setting and the initial values for each setting

Below is a view of the new addition to the Configuration module:
Screen Shot 2021-06-18 at 10 22 22 AM

#### Testing instructions 

1. Source the provided SQL patch
2. Check that the `MINC to BIDS Converter` settings appear in the Configuration module in your frontend
3. Check that submitting values for the related settings correctly updates the value in the frontend and backend

* Resolves #7477

### CODE REVIEW

File: SQL/New_patches/2021-06-17_mnc2bids_Config_Settings.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+--MINC to BIDS converter settings
+INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, Label, OrderNumber) VALUE ('minc2bids', 'Settings related to converting MINC to BIDS', 1, 0,  'MINC to BIDS Converter', 13);
```
Comment (cmadjar) [inline SQL/New_patches/2021-06-17_mnc2bids_Config_Settings.sql:2]:
```suggestion
INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, Label, OrderNumber) VALUES ('minc2bids', 'Settings related to converting MINC to BIDS', 1, 0,  'MINC to BIDS Converter', 13);
```

File: SQL/New_patches/2021-06-17_mnc2bids_Config_Settings.sql [hunk 2]
```diff
@@ -0,0 +1,12 @@
+--MINC to BIDS converter settings
```
Comment (cmadjar) [inline SQL/New_patches/2021-06-17_mnc2bids_Config_Settings.sql:1]:
```suggestion
-- MINC to BIDS converter settings
```

Not having the space make the patch impossible to source for some reason...

File: SQL/0000-00-03-ConfigTables.sql [hunk 3]
```diff
@@ -155,6 +155,13 @@ INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType,
 INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'modalities_to_deface', 'Modalities for which defacing should be run and defaced image inserted in the database', 1, 1, 'scan_type', ID, 'Modalities on which to run the defacing pipeline', 23 FROM ConfigSettings WHERE Name="imaging_pipeline";
 INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'default_bids_vl', 'Default visit label to use when no visit label set in the BIDS dataset', 1, 0, 'text', ID, 'Default visit label for BIDS dataset', 24 FROM ConfigSettings WHERE Name="imaging_pipeline";
 
+--MINC to BIDS converter settings
```
Comment (cmadjar) [inline SQL/0000-00-03-ConfigTables.sql:158]:
```suggestion
-- MINC to BIDS converter settings
```

Comment (cmadjar) [reply SQL/0000-00-03-ConfigTables.sql:158]: since the missing space made the patch fail, I guess it would make this fail as well

**Review by cmadjar — COMMENTED** [review]
@AlexandraLivadas I had issues running the patch and had to do the following to make it run.

I will be modifying the back end LORIS-MRI script that uses this config today and will let you know how things go.

Thank you :)

File: SQL/0000-00-03-ConfigTables.sql [hunk 4]
```diff
@@ -155,8 +155,8 @@ INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType,
 INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'modalities_to_deface', 'Modalities for which defacing should be run and defaced image inserted in the database', 1, 1, 'scan_type', ID, 'Modalities on which to run the defacing pipeline', 23 FROM ConfigSettings WHERE Name="imaging_pipeline";
 INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'default_bids_vl', 'Default visit label to use when no visit label set in the BIDS dataset', 1, 0, 'text', ID, 'Default visit label for BIDS dataset', 24 FROM ConfigSettings WHERE Name="imaging_pipeline";
 
---MINC to BIDS converter settings
-INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, Label, OrderNumber) VALUE ('minc2bids', 'Settings related to converting MINC to BIDS', 1, 0,  'MINC to BIDS Converter', 13);
+-- MINC to BIDS converter settings
+INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, Label, OrderNumber) VALUES ('minc2bids', 'Settings related to converting MINC to BIDS', 1, 0,  'MINC to BIDS Converter', 13);
```
Comment (cmadjar) [inline SQL/0000-00-03-ConfigTables.sql:159]:
```suggestion
INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, Label, OrderNumber) VALUES ('minc2bids', 'Settings related to converting MINC to BIDS LORIS-MRI tool script', 1, 0,  'MINC to BIDS Converter', 13);
```

File: SQL/0000-00-03-ConfigTables.sql [hunk 5]
```diff
@@ -155,6 +155,13 @@ INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType,
...
 INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'default_bids_vl', 'Default visit label to use when no visit label set in the BIDS dataset', 1, 0, 'text', ID, 'Default visit label for BIDS dataset', 24 FROM ConfigSettings WHERE Name="imaging_pipeline";
 
+-- MINC to BIDS converter settings
+INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, Label, OrderNumber) VALUES ('minc2bids', 'Settings related to converting MINC to BIDS', 1, 0,  'MINC to BIDS Converter', 13);
+INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'bids_dataset_authors', 'Authors for the BIDS dataset', 1, 1, 'text', ID, 'BIDS Dataset Authors', 1 FROM ConfigSettings WHERE Name='minc2bids';
+INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'bids_acknowledgments_text', 'Acknowledgments for the BIDS dataset', 1, 0, 'text', ID, 'BIDS Dataset Acknowledgments', 2 FROM ConfigSettings WHERE Name='minc2bids';
```
Comment (cmadjar) [inline SQL/0000-00-03-ConfigTables.sql:161]:
```suggestion
INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'bids_acknowledgments_text', 'Acknowledgments to be added in the dataset_description.json file of the BIDS dataset created out of the MINC files', 1, 0, 'text', ID, 'BIDS Dataset Acknowledgments', 2 FROM ConfigSettings WHERE Name='minc2bids';
```

File: SQL/0000-00-03-ConfigTables.sql [hunk 6]
```diff
@@ -155,6 +155,13 @@ INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType,
...
+INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'bids_readme_text', 'README for the BIDS dataset', 1, 0, 'textarea', ID, 'BIDS Dataset README', 3 FROM ConfigSettings WHERE Name='minc2bids';
```
Comment (cmadjar) [inline SQL/0000-00-03-ConfigTables.sql:162]:
```suggestion
INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'bids_readme_text', 'Content to be added to the README of the BIDS dataset generated out of the MINC files', 1, 0, 'textarea', ID, 'BIDS Dataset README', 3 FROM ConfigSettings WHERE Name='minc2bids';
```

File: SQL/0000-00-03-ConfigTables.sql [hunk 7]
```diff
@@ -155,6 +155,13 @@ INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType,
...
+INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'bids_validator_options_to_ignore', 'Options to ignore for BIDS validation', 1, 1, 'text', ID, 'BIDS Validation options to ignore', 4 FROM ConfigSettings WHERE Name='minc2bids';
```
Comment (cmadjar) [inline SQL/0000-00-03-ConfigTables.sql:163]:
```suggestion
INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'bids_validator_options_to_ignore', 'Options to be ignored for BIDS validation', 1, 1, 'text', ID, 'BIDS Validation options to ignore', 4 FROM ConfigSettings WHERE Name='minc2bids';
```

**Review by cmadjar — CHANGES_REQUESTED** [review]
@AlexandraLivadas I modified the LORIS-MRI script to be reading the new Config settings and it works great! Now I am at the documentation part and noticed that the description of some of the fields could be improved a little bit. I noted only the Patch but the same should be applied in the default schema.

Thank you!!

**Review by cmadjar — APPROVED** [review]
Looks great! Thank you @AlexandraLivadas !

**Comment by cmadjar** [comment]
@driusan ready for you :)

**Comment by driusan** [comment]
Merging because this is how we've done things so far, but in the future I'm not sure it makes sense to send SQL patches to LORIS with just inserts for config settings that only affect the LORIS-MRI side. Why can't the LORIS-MRI scripts just validate/do the insert statements themselves if they're missing? There are no special permissions required and that would make it easier to maintain instead of needing to send/synchronize PRs to multiple repositories.

**Pull request merged**